UndertaleModLib/Models/UndertaleTags.cs

```diff
@@ -7,71 +8,34 @@ namespace UndertaleModLib.Models;
         {
-            case UndertaleGameObject:
-                type = ResourceType.Object;
-                list = (IList<UndertaleNamedResource>) data.GameObjects;
-                break;
-            case UndertaleSprite:
-                type = ResourceType.Sprite;
-                list = (IList<UndertaleNamedResource>) data.Sprites;
-                break;
-            case UndertaleSound:
-                type = ResourceType.Sound;
-                list = (IList<UndertaleNamedResource>) data.Sounds;
-                break;
-            case UndertaleRoom:
-                type = ResourceType.Room;
-                list = (IList<UndertaleNamedResource>) data.Rooms;
-                break;
-            case UndertalePath:
-                type = ResourceType.Path;
-                list = (IList<UndertaleNamedResource>) data.Paths;
-                break;
-            case UndertaleScript:
-                type = ResourceType.Script;
-                list = (IList<UndertaleNamedResource>) data.Scripts;
-                offset = 100000;
-                break;
-            case UndertaleFont:
-                type = ResourceType.Font;
-                list = (IList<UndertaleNamedResource>) data.Fonts;
-                break;
-            case UndertaleTimeline:
-                type = ResourceType.Timeline;
-                list = (IList<UndertaleNamedResource>) data.Timelines;
-                break;
-            case UndertaleBackground:
-                type = ResourceType.Background;
-                list = (IList<UndertaleNamedResource>) data.Backgrounds;
-                break;
-            case UndertaleShader:
-                type = ResourceType.Shader;
-                list = (IList<UndertaleNamedResource>) data.Shaders;
-                break;
-            case UndertaleSequence:
-                type = ResourceType.Sequence;
-                list = (IList<UndertaleNamedResource>) data.Sequences;
-                break;
-            case UndertaleAnimationCurve:
-                type = ResourceType.AnimCurve;
-                list = (IList<UndertaleNamedResource>) data.AnimationCurves;
-                break;
+            UndertaleGameObject => ResourceType.Object,
+            UndertaleSprite => ResourceType.Sprite,
+            UndertaleSound => ResourceType.Sound,
+            UndertaleRoom => ResourceType.Room,
+            UndertalePath => ResourceType.Path,
+            UndertaleScript => ResourceType.Script,
+            UndertaleFont => ResourceType.Font,
+            UndertaleTimeline => ResourceType.Timeline,
+            UndertaleBackground => ResourceType.Background,
+            UndertaleShader => ResourceType.Shader,
+            UndertaleSequence => ResourceType.Sequence,
+            UndertaleAnimationCurve => ResourceType.AnimCurve,
+            _ => throw new ArgumentException("Invalid resource type!")
+        };
+        IList list = data[resource.GetType()] as IList;
```

Comment:
Not sure how I feel about this indexer....

Reply:
What's bad about it exactly?

Reply:
I personally just don't like is that it's not that clear from the outside how an indexer for the data file.

But it exists, and is used in other places, so might as well use it here, i guess.